### PR TITLE
Expose power on type to kernal

### PR DESCRIPTION
**Pull request by JimmyDansbo**
This change sets or resets a flag depending on if the system was powered on by a long-press on the powerbutton.
This is to give the kernal the opertunity to read this flag and jump into diagnostic mode if the system has been powered on by a long-press.
This change has no effect on an unmodified kernal.

### CODE REVIEW

File: x16-smc.ino [hunk 1]
```diff
@@ -77,6 +77,7 @@ bool SYSTEM_POWERED = 0;                                // default state - Power
 int  I2C_Data[3] = {0, 0, 0};
 bool I2C_Active = false;
 char echo_byte = 0;
+bool LONGPRESS_START = 0;	// Used to let CPU know NMI has come with pwr on
```
Comment (stefan-b-jakobsson) [inline x16-smc.ino:80]: Even though the C++ standard guarantees that true is 1 and false is 0, I don't think it is a good practice to rely on the conversion from bool to int and vice versa like is done in the code. If nothing else, it makes the code harder to read. Consider declaring the variable as uint8_t instead.

**Comment by stefan-b-jakobsson** [comment]
Apart from the comment above, the change looks good. If we want to reserve the last 512 bytes for a bootloader, there is very little space left in the flash memory for expanding the SMC firmware. But I don't think it's possible to do this using less memory.

Before merging this PR into the main branch we would need to know that the Kernal side is also approved.

**Comment by JimmyDansbo** [comment]
Thanks. change has been implemented. Still working on getting the kernal side of things completed.

**Comment by mooinglemur** [comment]
Seems like we might need to mirror those azduino.com assets or find another mirror.  This isn't the first time they've had extended problems.

**Comment by JimmyDansbo** [comment]
@mooinglemur I have added the file to my own server, if it helps: micronucleus-cli-2.5-azd1-x86_64-linux-gnu.tar.bz2
It does not seem like it gets updated very often. I found this on archive.org from august 2022

**Comment by stefan-b-jakobsson** [comment]
> Seems like we might need to mirror those azduino.com assets (for the CI build) or find another mirror. This isn't the first time they've had extended problems.

The server seems to be down. At least you can't ping it anymore, neither using the domain name nor the IP address. The whois looks OK, at least the domain is active. Maybe it comes up again, but you're right, they have their fair share of issues.

If we are to change where it's fetched from, I would prefer another kind of official source, if available. Otherwise we may store it in the x16-smc Github repo.

**Comment by mooinglemur** [comment]
I was able to flash my SMC with an external programmer using the asset out of this PR's action.
`avrdude -v -pattiny861 -cusbtiny -Uflash:w:firmware_with_bootloader.hex:i`

ROM side is merged.

**Comment by JimmyDansbo** [comment]
It seems the workflow is working fine again after micronucleus is become self-hosted :)

**Review by mooinglemur — APPROVED** [review]

**Pull request merged**